### PR TITLE
feat: configure CMC to play nice with cycles ledger

### PR DESCRIPTION
**Pull request by sesi200**

### CODE REVIEW

File: extensions/nns/src/install_nns.rs [hunk 1]
```diff
@@ -483,6 +486,61 @@ pub fn set_cmc_authorized_subnets(
...
+) -> anyhow::Result<()> {
+    #[derive(CandidType, Clone, Debug, PartialEq, Eq)]
+    struct CyclesCanisterInitPayload {
+        pub cycles_ledger_canister_id: Option<Principal>,
+    }
+
+    let wasm_path = nns_wasm_dir(dfx_cache_path);
+    let ledger_wasm_path = wasm_path.join(NNS_CYCLES_MINTING.wasm_name);
+    let ledger_wasm_bytes = std::fs::read(&ledger_wasm_path)
+        .with_context(|| format!("Failed to read {}", ledger_wasm_path.to_string_lossy()))?;
```
Comment (ericswanson-dfinity) [inline extensions/nns/src/install_nns.rs:503]:
```suggestion
    let ledger_wasm_bytes = dfx_core::fs::read(&ledger_wasm_path)?;
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

File: extensions-utils/src/dependencies/download_wasms/nns.rs [hunk 2]
```diff
@@ -152,6 +152,13 @@ pub const NNS_DAPP: StandardCanister = StandardCanister {
     wasm_name: "nns-dapp_local.wasm",
     wasm_url: "https://github.com/dfinity/nns-dapp/releases/download/tip/nns-dapp_t2.wasm",
 };
+/// Canister that can be used instead of cycle wallets
+pub const CYCLES_LEDGER: StandardCanister = StandardCanister {
+    canister_name: "cycles_ledger",
+    canister_id: "um5iw-rqaaa-aaaaq-qaaba-cai",
+    wasm_name: "cycles_ledger.wasm.gz",
+    wasm_url: "https://github.com/dfinity/cycles-ledger/releases/latest/download/cycles-ledger.wasm.gz",
```
[the remark this answers is not in the document]
Comment (ericswanson-dfinity) [reply extensions-utils/src/dependencies/download_wasms/nns.rs:160]: I've asked that this issue with II and nns-dapp be corrected in https://github.com/dfinity/dfx-extensions/pull/60

[the remark this answers is not in the document]
Comment (sesi200) [reply extensions-utils/src/dependencies/download_wasms/nns.rs]: Fair point. No, this is not needed, and we can make the update script keep this up to date if we ever want the extension to install the cylces ledger

File: extensions/nns/src/install_nns.rs [hunk 3]
```diff
@@ -483,6 +486,60 @@ pub fn set_cmc_authorized_subnets(
...
+    upgrade_arg_file
+        .write_all(
+            &Encode!(&(Some(CyclesCanisterInitPayload {
+                cycles_ledger_canister_id: Some(
+                    Principal::from_text(CYCLES_LEDGER.canister_id).unwrap()
+                ),
+            }),))
+            .unwrap(),
+        )
+        .unwrap();
```
Comment (ericswanson-dfinity) [inline extensions/nns/src/install_nns.rs:514]: the unwrap on Encode!() I can go along with, .write_all().unwrap() probably not

Comment (sesi200) [reply extensions/nns/src/install_nns.rs]: Agreed, and there were more mistakes. I must have dropped the self-review somehow

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**